source/dub/compilers/buildsettings.d

```diff
@@ -186,16 +168,27 @@ private:
 		assert(ary == ["path/foo.txt", "path2/foo2.txt"]);
 	}
 
-	static void removePaths(ref string[] arr, in string[] vals)
+	static bool pathMatch(string path, string pattern)
+	{
+		import std.functional : memoize;
+		
+		alias nativePath = memoize!((string stringPath) => NativePath(stringPath));
```

Comment:
It makes sense to update the `NativePath` implementation in `dub.internal` in the future to the one from vibe-core, which just stores the whole path string internally. Then this won't be necessary anymore.